### PR TITLE
Meeting class: User must now input time as well, not just date

### PR DESCRIPTION
**Pull request by jimseah-0116**
Meeting time is now compulsory

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindPolicyCommand.java [hunk 1]
```diff
@@ -16,8 +16,8 @@ public class FindPolicyCommand extends Command {
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Find person(s) with the input Policy Name linked to them.\n "
-            + "Parameters: pol/ (partial words are allowed and case-insensitive)\n"
-            + "Example: " + COMMAND_WORD + " pol/Saver ";
+            + "Parameters: [POLICY NAME] (partial words are allowed and case-insensitive)\n"
```
Comment (youdonnnn) [inline src/main/java/seedu/address/logic/commands/FindPolicyCommand.java:19]: Thanks for catching this error!

File: src/main/java/seedu/address/model/person/Meeting.java [hunk 2]
```diff
@@ -10,19 +10,14 @@
  * Guarantees: immutable; is always valid
  */
 public class Meeting implements Comparable<Meeting> {
-
+    private String name;
     public static final String MESSAGE_CONSTRAINTS =
-            "Meeting has to be in yyyy-MM-dd HH:mm:ss format, and it should not be blank";
-
+            "Meeting MUST be in yyyy-MM-dd HH:mm format, and it should be AFTER the current day and time.";
```
Comment (youdonnnn) [inline src/main/java/seedu/address/model/person/Meeting.java:15]: Good work fixing this bug :)

File: src/main/java/seedu/address/model/person/Meeting.java [hunk 3]
```diff
@@ -40,41 +35,41 @@ public LocalDateTime getMeeting() {
      * Returns true if a given string is a valid meeting.
      */
     public static boolean isValidMeeting(String test) {
-        return test.matches(VALIDATION_REGEX);
+        try {
+            return test.matches(VALIDATION_REGEX) && stringToDateTime(test).isAfter(LocalDateTime.now());
+        } catch (IllegalArgumentException e) {
+            return false;
+        }
```
Comment (rahhulleee) [inline src/main/java/seedu/address/model/person/Meeting.java:42]: Good to use a try catch block here

File: src/main/java/seedu/address/model/person/Meeting.java [hunk 4]
```diff
@@ -40,41 +35,41 @@ public LocalDateTime getMeeting() {
...
     }
 
     /**
      * This method converts a string into a LocalDateTime object.
-     * If the string is in the format "yyyy-MM-dd", it will be treated as a date at the start of the day.
-     * If the string is in the format "yyyy-MM-ddTHH:mm:ss", it will be treated as a date with time.
-     * Otherwise, the string is expected to be in the format "yyyy-MM-dd HH:mm:ss".
+     * The string must be in the format "yyyy-MM-dd HH:mm".
```
Comment (youdonnnn) [inline src/main/java/seedu/address/model/person/Meeting.java:47]: Not having to include the seconds during input is certainly more intuitive. Good work!

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 5]
```diff
@@ -22,22 +22,22 @@ public class SampleDataUtil {
     public static Person[] getSamplePersons() {
         return new Person[] {
             new Person(new Name("Alex Yeoh"), new Phone("87438807"), new Email("alexyeoh@example.com"),
-                new Address("Blk 30 Geylang Street 29, #06-40"), new Meeting("2024-02-12"),
+                new Address("Blk 30 Geylang Street 29, #06-40"), new Meeting("2028-02-12 10:00"),
```
Comment (youdonnnn) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:25]: Good work changing the sample data such that the address book can still load 👍

File: src/main/java/seedu/address/model/person/Meeting.java [hunk 6]
```diff
@@ -40,41 +35,41 @@ public LocalDateTime getMeeting() {
...
      */
-    private LocalDateTime stringToDateTime(String dateTime) {
-        if (dateTime.length() <= 10) {
-            return LocalDateTime.parse(dateTime + "T00:00:00", DateTimeFormatter.ISO_LOCAL_DATE_TIME);
-        } else if (dateTime.contains("T")) {
-            return LocalDateTime.parse(dateTime, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
-        } else {
-            return LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
+    private static LocalDateTime stringToDateTime(String dateTime) {
+        if (dateTime.length() != 16) {
```
Comment (youdonnnn) [inline src/main/java/seedu/address/model/person/Meeting.java:53]: LGTM! Removing the use of seconds in the datetime object seems to work better!

**Review by youdonnnn — APPROVED** [review]
LGTM! Thanks for the hard work.

File: src/main/java/seedu/address/model/person/Meeting.java [hunk 7]
```diff
@@ -40,41 +35,41 @@ public LocalDateTime getMeeting() {
...
+            throw new IllegalArgumentException("The date and time must be in the format yyyy-MM-dd HH:mm");
         }
+        return LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
     }
 
     public boolean isFutureMeeting() {
-        return this.meeting.isAfter(LocalDateTime.now())
-                || this.meeting.isEqual(LocalDateTime.now());
+        return this.meeting.isAfter(LocalDateTime.now()) ||
```
Comment (rahhulleee) [inline src/main/java/seedu/address/model/person/Meeting.java:60]: Thanks for fixing the checkstyle. It makes code more readable.

**Review by rahhulleee — APPROVED** [review]
Overall very good and necessary changes. Thanks jim. LGTM

File: src/main/java/seedu/address/model/person/Meeting.java (hunk 7 above)
Comment (Greg-Tan) [inline src/main/java/seedu/address/model/person/Meeting.java:60]: Great job fixing the format to improve code quality

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 8]
```diff
@@ -106,7 +106,7 @@ public static Meeting parseMeeting(String meeting) throws ParseException {
         requireNonNull(meeting);
         String trimmedMeeting = meeting.trim();
         if (!Meeting.isValidMeeting(trimmedMeeting)) {
-            throw new ParseException(Email.MESSAGE_CONSTRAINTS);
+            throw new ParseException(Meeting.MESSAGE_CONSTRAINTS);
```
Comment (irwynliong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:109]: good catch of error!

File: src/main/java/seedu/address/model/person/Meeting.java [hunk 9]
```diff
@@ -40,41 +35,41 @@ public LocalDateTime getMeeting() {
...
+            this.meeting.isEqual(LocalDateTime.now());
     }
-
+    
+
```
Comment (irwynliong) [inline src/main/java/seedu/address/model/person/Meeting.java:64]: could remove these white spaces

File: src/main/java/seedu/address/model/person/Meeting.java [hunk 10]
```diff
@@ -40,41 +35,41 @@ public LocalDateTime getMeeting() {
...
+    
     public void setName(String finalName) {
         this.name = finalName;
     }
 
     public String getName() {
         return this.name;
     }
-
+
```
Comment (irwynliong) [inline src/main/java/seedu/address/model/person/Meeting.java:72]: checkstyle issue here

**Review by irwynliong — APPROVED** [review]
a few checkstyle issue but LGTM!

**Pull request merged**